### ISSUE TITLE
Missing minimum CMake version in Hologram demo

### ISSUE THREAD

**Issue by SaschaWillems** [LunarG/VulkanSamples#45]
The [CMakeLists.txt](https://github.com/LunarG/VulkanSamples/blob/master/Demos/Hologram/CMakeLists.txt) in the hologram demo is not specifiying a minimum cmake version (the other examples do), resulting in the following warning : 

	CMake Warning (dev) in CMakeLists.txt:
	  No cmake_minimum_required command is present.  A line of code such as

		cmake_minimum_required(VERSION 3.1)

	  should be added at the top of the file.  The version specified may be lower
	  if you wish to support older CMake versions for this project.  For more
	  information run "cmake --help-policy CMP0000".
	This warning is for project developers.  Use -Wno-dev to suppress it.

**Comment by olvaffe** [comment]
Hologram relies on the top-level CMakeLists.txt to perform the check.  Is it advised each subdirectory perform the check itself?

Or were you trying to run cmake from within Hologram?  That would make sense, but Linux build would still fail because FindXCB.cmake would be missing.

**Comment by SaschaWillems** [comment]
> Or were you trying to run cmake from within Hologram? That would make sense, but Linux build would still fail because FindXCB.cmake would be missing.

Exactly. I only wanted to compile the Hologram demo without having to build all dependencies for the other examples, so I used it's CMakeLists.txt. On windows this works fine.

**Comment by olvaffe** [comment]
There are common setups (FindXCB.cmake included) that cannot be found in Hologram's CMakeLists.txt.  I guess the right way here is for the top-level CMakeLists.txt to make samples and their dependencies optional.

**Comment by ChristophHaag** [comment]
FindXCB.cmake is in extra-cmake-modules

According to http://api.kde.org/ecm/manual/ecm-find-modules.7.html adding
```
find_package(ECM REQUIRED NO_MODULE)
set(CMAKE_MODULE_PATH ${ECM_FIND_MODULE_DIR})
```
in front of `find_package(XCB REQUIRED)` lets cmake find it.

I also had to add `set_property(TARGET Hologram PROPERTY CXX_STANDARD 11)` because this cmakelists file doesn't set c++11.

Oh and from the Hologram directory, you also need to put a glslang build e.g. `ln -s /wherever/glslang ../../../glslang`.

With all that I could build it "standalone".
Tl;dr: https://github.com/ChristophHaag/VulkanSamples/commit/c491ec997095efce68f5dd85a9f58fe8065f412c

Does LunarG want to support "standalone" builds of the samples like that?

**Comment by Tony-LunarG** [comment]
Running CMake at the root of repo to create all the makefiles and then going to the Hologram directory and running make builds just Hologram - is that sufficient?  I don't think we need to support running CMake in subdirectories

**Comment by Tony-LunarG** [comment]
Also, after running CMake from the root of the repo, running "make Hologram" will also just make the Hologram sample

**Comment by Tony-LunarG** [comment]
Closing - please open a new issue if there is still a problem
